### ISSUE TITLE
osu! will minimize when trying to open, taking multiple attempts for it to actually open

### ISSUE THREAD

**Issue by rymeilema** [ppy/osu-stable-issues#323]
after osu! is already opened in fullscreen, i tend to minimize it a lot, to chat in discord or to download new beatmaps, etc
when trying to maximize it again, it just minimizes, requiring click spam on the tray icon for it to actually maximize
this has been going on for about 2 months - #help suggested it was outdated graphic drivers, but mine are updated
none of the many recent stable updates have helped whatsoever
alt+enter to make it windowed doesn't have this problem, but alt+enter to make it fullscreen again will not place osu! on front, as it also minimizes

no files attached because screen recording software can't capture it, will provide anything else needed

**Comment by peppy** [comment]
If possible, could you please take a video using a smartphone or otherwise?

**Comment by Peter-GHub** [comment]
it happens on compatibility mode for me as well 
Here's short video gif showing this problem
https://gyazo.com/e5130c00485dd3477aeb97f8fb868084

It doesn't happen with compatibility mode off from what i've seen

**Comment by peppy** [comment]
if it's only with compatibility on we will not be fixing this.

**Comment by Nyrdie** [comment]
So this recently started happening, whenever I minimize the osu! window once, I click on it to maximize it again, but it ends up abruptly minimizing over and over again. Although there were instances where it would maximize as normal, most of the time the former ends up being the case. I haven't really messed with the compatibility settings, as far as I know it just started happening at a certain point. Hopefully it isn't a virus thing, and can be fixed. Thank you for seeing this.

https://user-images.githubusercontent.com/77835966/105483943-8f842380-5ce5-11eb-80f3-30ca9f613089.mp4

**Comment by PapiDieter69** [comment]
I have the exact same problem that Nyrdie shows in the video.

**Comment by PapiDieter69** [comment]
I had the exact same problem that Nyrdie. It turns out that i somehow activated compatibility mode and after deactivating it the problem was gone.

**Comment by peppy** [comment]
Thanks for the video, but we need to know a lot more to actually take any kind of action:

- What GPU are you using?
- What driver version are you using?
- Does rolling back the drivers fix your issue?
- Does this only happen when in fullscreen mode?

**Comment by Nyrdie** [comment]
So my pc currently uses an Nvidia GTX 1050 Ti. If you're referring to the GeForce driver, I am currently updated with the latest game ready version "461.09". Sadly, I haven't tried rolling back my drivers because I don't know how to. And lastly yes, as I've tried it, it only happens when I am in fullscreen mode, rendering at native resolution or not.

**Comment by smoogipoo** [comment]
Can replicate this one locally. Please provide your `runtime.log` file.

(I believe this is D3D device loss on minimise, at an ANGLE level, but definitely NV-driver-related).

**Comment by Nyrdie** [comment]
I am sorry to have bothered you, but apparently I may have turned on the compatibility setting a long time ago and forgot that I turned it on because it never showed any problems until now. I have turned it off, and it works fine now. For everyone who has the same problem, kindly check the compatibility mode under the GRAPHICS settings. Whoops.

**Comment by greevak** [comment]
I use stable, and i get the same issue. Gabixel did mention it so there you go.

**Comment by peppy** [comment]
Thanks, but there is no need to comment here unless you have more details to add to the issue.

**Comment by K41t0uK1dd0** [comment]
Having the same issue here as well. Also https://github.com/ppy/osu/issues/11598
![image](https://user-images.githubusercontent.com/22808988/148666787-019b8a97-72e6-4ac4-9e3b-24227c3ffc63.png)

**Comment by smoogipoo** [comment]
This is ANGLE/compatibility mode related. We don't have a fix for the time being, pending more work in osu!lazer first.

**Comment by XKMR** [comment]
the problem is not fixed yet. as a new player, I'm really confused why they are not fixing it.

**Comment by peppy** [comment]
@XKMR are you running in compatibility mode?

**Comment by XKMR** [comment]
well I turned it off and it's now kind of better. it still bugs sometimes though, not a problem anymore. also what is the use of the compatibility mode?

**Comment by nikitOS4323** [comment]
disabling fullscreen optimizations solved the issue for me

**Comment by karyusama** [comment]
I have the same issue while trying to fix stuttering. I have to use the Vsync option because I had screen tearing and with the Vsync option it goes away but I have 14 ms, wich it's not the best and with compatibility mode I have 2 ms, it's a huuuge difference but when I alt-tab I get this black screen flashes. I don't know if there's a way I can fix the stuttering and screen tearing. I can record it if needed but is the same as in Nyrdie's video

**Comment by sl1nkZz** [comment]
Yup same for me, disabling fullscreen optimizations solved the issue. I'm running on Win10 Pro (2h22, Build: 19045.3324) and I'm playing osu in compatibility mode.

My specs:

MSI MAG B550 GAMING PLUS
AMD Ryzen 7 5800X
AMD Radeon RX 6800XT
Win10 (2h22, Build 19045.3324)

I can tab in and out without problems now.
